### PR TITLE
handle env variables in dynamic secret file

### PR DESCRIPTION
**Pull request by alban-stourbe-wmx**
## Proposed changes
Add env variables  in dynamic secret files. 

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [ ] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: internal/runner/lazy.go [hunk 1]
```diff
@@ -75,7 +77,16 @@ func GetLazyAuthFetchCallback(opts *AuthLazyFetchOptions) authx.LazyFetchSecret
 		vars := map[string]interface{}{}
 		mainCtx := context.Background()
 		ctx := scan.NewScanContext(mainCtx, contextargs.NewWithInput(mainCtx, d.Input))
+
 		for _, v := range d.Variables {
+			//  Check if the template has any env variables and expand them
+			if strings.HasPrefix(v.Value, "$") {
```
Comment (dogancanbakir) [inline internal/runner/lazy.go:83]: we can use https://github.com/projectdiscovery/utils/blob/main/env/env.go#L17

**Review by dogancanbakir — CHANGES_REQUESTED** [review]

**Review by tarunKoyalwar — APPROVED** [review]
lgtm ! 

along with env secrets now secrets can also be specified from -v and -env-vars ex:

### Env based secret
```yaml
variables:
   - key: password
     value: $PASSWORD
```

### Config file based secrets ( using -env-vars or -vars )
```yaml
variables:
     - key: password
     - value: {{password}}
```

**Comment by alban-stourbe-wmx** [comment]
> * needs rebase from dev

Hi @tarunKoyalwar, the rebase has been made. Thank for the code added ☝️

**Review by ehsandeep — APPROVED** [review]
Thank you @alban-stourbe-wmx for this update, FYI we are also working on https://github.com/projectdiscovery/nuclei/pull/5841

**Comment by alban-stourbe-wmx** [comment]
> Thank you @alban-stourbe-wmx for this update, FYI we are also working on #5841

Thanks to you and also congrats to @dwisiswant0 and @tarunKoyalwar

**Pull request merged**